packages/cli/src/commands/components/pull/actions.ts

```diff
@@ -24,11 +27,17 @@ export const fetchComponent = async (space: string, componentName: string): Prom
   try {
     const client = mapiClient();
 
-    const { data } = await client.get<{
-      components: SpaceComponent[];
-    }>(`spaces/${space}/components?search=${encodeURIComponent(componentName)}`, {
+    const { data } = await client.components.list({
+      path: {
+        space_id: Number.parseInt(space, 10),
```

Comment:
suggestion (non-blocking): I've noticed that `space_id` allows both `string` and `number`, so parsing it is not strictly necessary here.

General side note: in cases where we want to transform a string that must be a valid number to a number (e.g. `"123"`) `Number(string)` is often preferable to `parseInt()`. `parseInt` is the right solution when dealing with messy input (e.g. `"20px"`).